### DISCUSSION TITLE
Coordinate transformation for displaying

### DISCUSSION THREAD

**Discussion by aanastasiou** [mozman/ezdxf#492] [Q&A] [answered]
Hello and thank you very much for ezdxf.

Is there any special transformation that has to be applied to the entity coordinates as they are reported via ezdxf before using them?

I noticed that there are a few different coordinate systems (WCS, OCS, DCS) as well as a couple of viewports defined in the file I am interested in. However, I can produce a very quick rendering of the entities of interest via a very simple linear transform, e.g. `(x-x_min)/(x_max-x_min)` (and also for `y`) would bring my entities within `0..1` and then from there (and taking the aspect ratio of this bounding box as well) produce a very simple outline preview of the contents of a file.

The question now is, are there any additional transformations that must be applied to the coordinates, as part of the way they are stored within the DXF file before they are considered "correct"? I noticed things such as "extrusion vector" (?), that I am in the process of reading about but if a simple linear transform is enough to "draw" the points on a plain pixel surface then I would use that. (Not an Autocad user myself either).

All the best

**Comment by mozman** [comment, marked as answer]
The OCS to WCS transformation if the extrusion vector != (0, 0, 1), see https://ezdxf.mozman.at/docs/concepts/ocs.html

```Python
# example for OCS entity CIRCLE
ocs = circle.ocs()
wcs_center = ocs.to_wcs(circle.dxf.center)
```

**Reply by aanastasiou** [reply]
That's good to know, thank you very much (all of my entities of interest are 2D on this occasion).

**Reply by mozman** [reply]
Sometimes an extrusion vector of (0, 0, -1) can occur for transformed entities (reflections! - especially if transformed by ezdxf) even for 2D entities.

**Reply by aanastasiou** [reply]
@mozman thanks, will be keeping an eye on things like this.
